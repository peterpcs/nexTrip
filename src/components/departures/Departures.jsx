import React from 'react'
import StopDescription from './StopDescription'
import StopDepartures from './StopDepartures'

export default function Departures(props) {
  return (
    <div id="nextripDepartures" className="nextrip-departures">
      <div className="card mb-3 p-3 p-md-3">
        <StopDescription {...props} />
        <StopDepartures {...props} />
      </div>
    </div>
  )
}
