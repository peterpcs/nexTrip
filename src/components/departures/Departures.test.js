import * as React from 'react'
import '@testing-library/jest-dom/extend-expect'
import Departures from './Departures'
import renderer from 'react-test-renderer'

const nextTrips = {
  stops: [
    {
      stop_id: 51405,
      latitude: 44.854277,
      longitude: -93.238877,
      description: 'MOA Transit Station',
    },
  ],
  alerts: [],
  departures: [
    {
      actual: true,
      trip_id: '20521637-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '11 Min',
      departure_time: 1652725200,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521638-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '1:32',
      departure_time: 1652725920,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521636-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '1:44',
      departure_time: 1652726640,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521639-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '1:56',
      departure_time: 1652727360,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521640-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '2:08',
      departure_time: 1652728080,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521641-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '2:20',
      departure_time: 1652728800,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521642-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '2:32',
      departure_time: 1652729520,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521643-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '2:44',
      departure_time: 1652730240,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521644-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '2:56',
      departure_time: 1652730960,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
    {
      actual: false,
      trip_id: '20521645-MAR22-RAIL-Weekday-01',
      stop_id: 51405,
      departure_text: '3:08',
      departure_time: 1652731680,
      description: 'to Mpls-Target Field',
      route_id: '901',
      route_short_name: 'Blue',
      direction_id: 0,
      direction_text: 'NB',
      schedule_relationship: 'Scheduled',
    },
  ],
}

test('renders departures correctly', () => {
  const tree = renderer.create(<Departures nextTrips={nextTrips} />).toJSON()
  expect(tree).toMatchSnapshot()
})
