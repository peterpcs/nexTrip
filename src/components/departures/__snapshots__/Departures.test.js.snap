// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders departures correctly 1`] = `
<div
  className="nextrip-departures"
  id="nextripDepartures"
>
  <div
    className="card mb-3 p-3 p-md-3"
  >
    <div
      className="stop-description mb-3"
    >
      <h3
        className="h2 stop-name"
      >
        MOA Transit Station
      </h3>
      <span
        className="stop-number"
      >
        <strong>
          Stop #: 
        </strong>
        51405
      </span>
    </div>
    <div
      className="stop-departures"
    >
      <table
        className="table departures-table"
      >
        <thead>
          <tr>
            <th
              className="route"
            >
              Route
            </th>
            <th
              className="destination"
            >
              Destination
            </th>
            <th
              className="departs text-right"
            >
              Departs
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <img
                alt="broadcast icon"
                className="icon blink mr-1"
                src="broadcast-blue.svg"
              />
              <span>
                11 Min
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                1:32
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                1:44
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                1:56
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                2:08
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                2:20
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                2:32
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                2:44
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                2:56
              </span>
            </td>
          </tr>
          <tr
            className="departure"
            style={
              Object {
                "display": "table-row",
              }
            }
          >
            <td
              className="route-number mr-2"
            >
              Blue
            </td>
            <td
              className="route-name"
            >
              to Mpls-Target Field
            </td>
            <td
              className="depart-time ml-auto"
            >
              <span>
                3:08
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
